spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala

```diff
@@ -1081,9 +1081,21 @@ object VacuumProtocolCheckTableFeature
 object CheckpointProtectionTableFeature
     extends WriterFeature(name = "checkpointProtection")
     with RemovableFeature {
+  /**
+   * Gets the version requiring checkpoint protection from `metadata`. If the table property is
+   * not set, return `None`.
+   */
+  def getCheckpointProtectionVersionOption(protocol: Protocol, metadata: Metadata): Option[Long] = {
```

Comment:
This is a different refactor. Can you update the PR description and title to reflect this change too?